frame/nomination-pools/src/lib.rs

```diff
@@ -411,6 +411,21 @@ enum AccountType {
 	Reward,
 }
 
+/// Account to bond pending reward of a member.
+#[derive(Encode, Decode, MaxEncodedLen, Clone, Copy, Debug, PartialEq, Eq, TypeInfo)]
+pub enum ClaimPermission {
+	/// Only the pool member themself can claim their reward.
+	Permissioned,
+	/// Anyone can claim the reward of this member.
+	Permissionless,
```

Comment:
Actually, now I think we should let this be fine grained as well: 
```
enum ClaimPermission { 
   Permissioned,
   PermissionlessAll,
   PermissionlessCompund,
   PermissionlessWithdraw,
}
```

WDYT @rossbulat? don't want to stretch this any longer, but I think this might be better.

Reply:
I like, I will commit some updates shortly. Additional call `claim_payout_other` needed for the `PermissionlessWithdraw`.